### ISSUE TITLE
SQLPro for SQLite v 2021.27 - Crasher and Freezer

### ISSUE THREAD

**Issue by Maximost** [hankinsoft/SQLPro#713]
Good afternoon.

I have a couple of issues that I've been experiencing while running SQLPro for SQLite on a Macintosh under the Catalina OS. Both of these issues have been occurring while editing a small SQLite database

1. Unexpectedly quits when adding rows to tables. Sometimes can add several rows before the app exits, other times adding only 1 row will cause the app to quit.
2. When exporting a CSV file, the app will complete the export but the export progress bar does not close. The app can no longer be accessed. Must force-quit to exit the app.

Really useful application but the quits and freezes make using it frustrating.

**Comment by hankinsoft** [comment]
Thanks, I'm working on getting a new website build available. If possible, could you grab https://apps.apple.com/us/app/sqlpro-for-sqlite/id586001240?mt=12 using the free promo code T47H6E9YYMNP
and let me know if that sorts the issue?

**Comment by hankinsoft** [comment]
The website version https://d3fwkemdw8spx3.cloudfront.net/sqlite/SQLProSQLite.2021.110.app.zip is also now available.

**Comment by Maximost** [comment]
Thank you for sending the new version. I was able to export CSV files without any issues. In other words, the export progress bar dismissed itself politely, csv files were created and saved and the program was ready for my next task.

Adding rows to a table using the + button at the bottom of the interface still causes the program to quit without warning. Adding rows is a bit less of an issue for me at least because I don't necessarily need to do this on a regular basis. Of course, that's only my perspective.

**Comment by hankinsoft** [comment]
> Adding rows to a table using the + button at the bottom of the interface still causes the program to quit without warning. Adding rows is a bit less of an issue for me at least because I don't necessarily need to do this on a regular basis. Of course, that's only my perspective.

Still investigating. I thought I had fixed this, but I was just able to reproduce it 1 out of ~100 times. Hoping I can reproduce it again.

**Comment by hankinsoft** [comment]
I believe I have this fixed. If you get the chance, could you give:
https://d3fwkemdw8spx3.cloudfront.net/sqlite/SQLProSQLite.2022.2.app.zip

A try?

**Comment by Maximost** [comment]
Thank you so much for the updated version.

Yes, it looks like the unexpected quit when adding rows to a table has been fixed. I opened an existing database that has a lot of data and tried adding and removing rows from several different tables. The first table has three columns, the second table 10 columns, and the third around 20 columns. Each table has a mix of data types declared for the columns (int, char, float, bool). I was able to add 40+ rows to each table by repeatedly clicking on the add row button without any issues. I then deleted rows individually using the row delete button and also selected multiple rows and deleted them using the row delete button. No problems whatsoever.

I’ll be doing some data merging later next week, so will have a chance to use the add and remove row features a little more. I’ll let you know if the bug returns.
